src/User32.Desktop/User32.cs

```diff
@@ -57,7 +56,61 @@ public static partial class User32
         /// <summary>Windows icon or the icon of the window specified in <see cref="MENUITEMINFO.dwItemData" />.</summary>
         public static readonly IntPtr HBMMENU_SYSTEM = new IntPtr(1);
 
-        public delegate int WindowsHookDelegate(int code, IntPtr wParam, IntPtr lParam);
+        /// <summary>
+        /// An application-defined or library-defined callback function used with the <see cref="SetWindowsHookEx(WindowsHookType, IntPtr, IntPtr, int)"/> function.
+        /// This is a generic function to Hook callbacks. For specific callback functions see CallWndDelegate, CallWndRetDelegate, CBTDelegate, DebugDelegate, ForegroundIdleDelegate,
+        /// GetMsgDelegate, JournalPlaybackDelegate, JournalRecordDelegate, KeyboardDelegate, LowLevelKeyboardDelegate, MessageDelegate, MouseDelegate, LowLevelMouseDelegate,
+        /// ShellDelegate and SysMsgDelegate.
```

Comment:
Nope, this was my contribution to the documentation before our discussion on that and opening the Issue, I guess the proper thing would be to take out the references because I understood the best is to keep only the IntPtr and the generic delegate. I think I will update that just to cite that there are specializations and that it should be checked on MSDN for more details on that.